src/librbd/librbd.cc

```diff
@@ -2552,6 +2582,25 @@ namespace librbd {
     return r;
   }
 
+  int Image::diff_iterate3(const int64_t group_pool,
+                          const std::string &group_id,
+                          const std::string &group_snapshot_id,
+                          const char *fromsnapname, uint64_t ofs, uint64_t len,
+                          bool include_parent, bool whole_object,
+                          int (*cb)(uint64_t, size_t, int, void *), void *arg)
```

Comment:
This looks rather confusing to me. What about having something like:
```
int Image::diff_iterate_group_snap(const char *fromsnapname, uint64_t ofs, uint64_t len,
￼                          bool include_parent, bool whole_object,
￼                          int (*cb)(uint64_t, size_t, int, void *), void *arg)
```
Where `fromsnapname` would be the group snapshot name. And using it, making some additional low level calls, you would find all necessary information to make `cls::rbd::GroupSnapshotNamespace` and the image `fromsnapname` to call `librbd::api::DiffIterate<>::diff_iterate`?

Reply:
@trociny  The problem has been revised as required